pkg/segment/aggregations/evalaggs.go

```diff
@@ -769,6 +771,123 @@ func ComputeAggEvalForList(measureAgg *structs.MeasureAggregator, sstMap map[str
 	return nil
 }
 
+func ComputeAggEvalForPerc(measureAgg *structs.MeasureAggregator, sstMap map[string]*structs.SegStats, measureResults map[string]sutils.CValueEnclosure, runningEvalStats map[string]interface{}) error {
+	fields := measureAgg.ValueColRequest.GetFields()
+	fieldToValue := make(map[string]sutils.CValueEnclosure)
+	// result := 0.0
+	var err error
+	var td *utils.GobbableTDigest
+	_, ok := runningEvalStats[measureAgg.String()]
+	if !ok {
+		td, err = utils.CreateNewTDigest()
+		if err != nil {
+			return fmt.Errorf("ComputeAggEvalForPerc: can not create a new digest tree; measureAgg: %v, err: %v", measureAgg.String(), err)
+		}
+		runningEvalStats[measureAgg.String()] = td
+	} else {
+		td, ok = runningEvalStats[measureAgg.String()].(*utils.GobbableTDigest)
+		if !ok {
+			return fmt.Errorf("ComputeAggEvalForPerc: can not convert to utils.GobbableTDigest for measureAgg: %v; err: %v", measureAgg.String(), err)
+		}
+	}
+
+	if len(fields) == 0 {
+		countStat, exist := sstMap["*"]
+		if !exist {
+			return fmt.Errorf("ComputeAggEvalForPerc: sstMap did not have c")
+		}
+		err = PerformEvalAggForPerc(measureAgg, countStat.Count, td, fieldToValue)
+		if err != nil {
+			return fmt.Errorf("PerformEvalAggForPerc: Error while performing eval agg for perc, err: %v", err)
+		}
+	} else {
+		sst, ok := sstMap[fields[0]]
+		if !ok {
+			return fmt.Errorf("ComputeAggEvalForPerc: sstMap did not have segstats for field %v, measureAgg: %v", fields[0], measureAgg.String())
+		}
+		length := len(sst.Records)
```

Comment:
nit: this is more clear
```suggestion
		numRecords := len(sst.Records)
```